sources/packages/backend/libs/integrations/src/services/restriction/student-restriction.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from "@nestjs/common";
+import { NotificationActionsService } from "@sims/services/notifications";
+import {
+  RecordDataModelService,
+  RestrictionActionType,
+  StudentRestriction,
+} from "@sims/sims-db";
+import {
+  ArrayContains,
+  DataSource,
+  EntityManager,
+  In,
+  Not,
+  SelectQueryBuilder,
+} from "typeorm";
+export const RESTRICTION_NOT_ACTIVE = "RESTRICTION_NOT_ACTIVE";
+export const RESTRICTION_NOT_PROVINCIAL = "RESTRICTION_NOT_PROVINCIAL";
+/**
+ * While performing a possible huge amount of select,
+ * breaks the execution in chunks.
+ */
+const NOTIFICATIONS_SELECT_CHUNK_SIZE = 1000;
+
+/**
+ * Service layer for Student Restriction.
+ */
+@Injectable()
+export class IntegrationStudentRestrictionService extends RecordDataModelService<StudentRestriction> {
```

Comment:
It seems to be used across many layers, is it not more suitable to be moved to `@sims/services`?